redisextensions/redis.go

```diff
@@ -0,0 +1,93 @@
+package redisextensions
+
+/*
+ * Copyright (c) 2021 TFG Co <backend@tfgco.com>
+ * Author: TFG Co <backend@tfgco.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/spf13/viper"
+	redistracing "github.com/topfreegames/extensions/v9/tracing/redis/v8"
+)
+
+type ClientConfig struct {
+	URL string
+	// Timeout in seconds
+	ConnectionTimeout int
```

Comment:
You can change this to `time.Duration` and execute a viper.GetDuration instead.

Reply:
I would like to keep it that way so we don't break semantic compatibility.

For instance, this old code `redis.NewClientFromConfig(ctx, &redis.ClientConfig{URL: "http://localhost", ConnectionTimeout: 60})` would still compile after the upgrade, however, the end result would be different.

This kind of problem would be hard to catch.

Reply:
Right! But see, this kind of change is possible now that you are making a major change.

Since you created a new directory for this version, you'll already need to change your code to use the new import path `github.com/topfreegames/extensions/redis/v8` (this is the point of creating a version directory, indeed). So why not use this opportunity to have a more idiomatic code?

Reply:
Done